File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/CompositeFuture.php
<?php 
namespace io\vertx\php\core;
class CompositeFuture
{
    
    private function __construct()
    {

    }
    /**
     * (Future f1, Future f2)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4, Future f5)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4, Future f5, Future f6)
     * @return CompositeFuture
     * (string futures)
     * @return CompositeFuture
     * methodSize : 6
     */
    public function all(/*6*/$arg0, $arg1, $arg2, $arg3, $arg4, $arg5)
    {

    }
    /**
     * (Future f1, Future f2)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4, Future f5)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4, Future f5, Future f6)
     * @return CompositeFuture
     * (string futures)
     * @return CompositeFuture
     * methodSize : 6
     */
    public function any(/*6*/$arg0, $arg1, $arg2, $arg3, $arg4, $arg5)
    {

    }
    /**
     * (int index)
     * @return Throwable
     * methodSize : 1
     */
    public function cause(/*1*/$arg0)
    {

    }
    /**
     * (CompositeFuture result)
     * @return Void
     * ()
     * @return Void
     * methodSize : 2
     */
    public function complete(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return callable
     * methodSize : 1
     */
    public function completer(/*0*/)
    {

    }
    /**
     * (callable handler, Future next)
     * @return Future
     * (callable mapper)
     * @return Future
     * methodSize : 2
     */
    public function compose(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (int index)
     * @return bool
     * methodSize : 1
     */
    public function failed(/*1*/$arg0)
    {

    }
    /**
     * (int index)
     * @return bool
     * methodSize : 1
     */
    public function isComplete(/*1*/$arg0)
    {

    }
    /**
     * (Future f1, Future f2)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4, Future f5)
     * @return CompositeFuture
     * (Future f1, Future f2, Future f3, Future f4, Future f5, Future f6)
     * @return CompositeFuture
     * (string futures)
     * @return CompositeFuture
     * methodSize : 6
     */
    public function join(/*6*/$arg0, $arg1, $arg2, $arg3, $arg4, $arg5)
    {

    }
    /**
     * (callable mapper)
     * @return Future
     * methodSize : 1
     */
    public function map(/*1*/$arg0)
    {

    }
    /**
     * (callable mapper)
     * @return Future
     * (CompositeFuture value)
     * @return Future
     * methodSize : 2
     */
    public function otherwise(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return Future
     * methodSize : 1
     */
    public function otherwiseEmpty(/*0*/)
    {

    }
    /**
     * (callable mapper)
     * @return Future
     * methodSize : 1
     */
    public function recover(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return CompositeFuture
     * methodSize : 1
     */
    public function result(/*0*/)
    {

    }
    /**
     * (int index)
     * @return Memory
     * methodSize : 1
     */
    public function resultAt(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return CompositeFuture
     * methodSize : 1
     */
    public function setHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return int
     * methodSize : 1
     */
    public function size(/*0*/)
    {

    }
    /**
     * (int index)
     * @return bool
     * methodSize : 1
     */
    public function succeeded(/*1*/$arg0)
    {

    }
    /**
     * (CompositeFuture result)
     * @return bool
     * ()
     * @return bool
     * methodSize : 2
     */
    public function tryComplete(/*1*/$arg0)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/CompositeFuture.java
package io.vertx.php.core;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.Future;
@Name("CompositeFuture")
@Namespace("io\\vertx\\php\\core")
public interface CompositeFuture<S extends io.vertx.core.CompositeFuture> extends IWrapper<S>{
    
    @Signature
    default Memory isComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isComplete()
    }

    @Signature
    default Memory isComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isComplete()
    }
    
    @Signature
    default void complete(Environment __env__, Memory... args) {
        this.getWrappedObject().complete()
    }

    @Signature
    default void complete(Environment __env__, Memory... args) {
        this.getWrappedObject().complete()
    }

    @Signature
    default void fail(Environment __env__, Memory... args) {
        this.getWrappedObject().fail(cause)
    }
    
    @Signature
    default Memory tryComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().tryComplete()
    }

    @Signature
    default Memory tryComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().tryComplete()
    }

    @Signature
    default Memory tryFail(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().tryFail(cause)
    }
    
    @Signature
    default Memory result(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, CompositeFutureImpl::new, this.getWrappedObject().result()
    }
    
    @Signature
    default Memory cause(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().cause()
    }

    @Signature
    default Memory cause(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().cause()
    }
    
    @Signature
    default Memory succeeded(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().succeeded()
    }

    @Signature
    default Memory succeeded(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().succeeded()
    }
    
    @Signature
    default Memory failed(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().failed()
    }

    @Signature
    default Memory failed(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().failed()
    }

    @Signature
    default Memory compose(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().compose(mapper)
    }

    @Signature
    default Memory map(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().map(mapper)
    }
    
    @Signature
    default Memory mapEmpty(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().mapEmpty()
    }
    
    @Signature
    default Memory completer(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().completer()
    }

    @Signature
    default Memory recover(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().recover(mapper)
    }

    @Signature
    default Memory otherwise(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().otherwise(mapper)
    }
    
    @Signature
    default Memory otherwiseEmpty(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().otherwiseEmpty()
    }

    @Signature
    static Memory all(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, CompositeFutureImpl::new, io.vertx.core.CompositeFuture.all(futures)
    }

    @Signature
    static Memory any(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, CompositeFutureImpl::new, io.vertx.core.CompositeFuture.any(futures)
    }

    @Signature
    static Memory join(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, CompositeFutureImpl::new, io.vertx.core.CompositeFuture.join(futures)
    }

    @Signature
    default Memory setHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().setHandler(handler)
    }

    @Signature
    default Memory resultAt(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().resultAt(index)
    }
    
    @Signature
    default Memory size(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().size()
    }

}
